extension/basicauthextension/extension_test.go

```diff
@@ -193,3 +206,108 @@ func TestBasicAuth_SupportedHeaders(t *testing.T) {
 		assert.NoError(t, err)
 	}
 }
+
+type mockRoundTripper struct{}
+
+func (m *mockRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp := &http.Response{StatusCode: http.StatusOK, Header: map[string][]string{}}
+	for k, v := range req.Header {
+		resp.Header[k] = v
+	}
+	return resp, nil
+}
+
+func TestBasicAuth_ClientValid(t *testing.T) {
+	creds := "username:password"
+	ext, err := newExtension(&Config{
+		Htpasswd: HtpasswdSettings{
+			Inline: creds,
+		},
+	})
+	assert.NotNil(t, ext)
```

Comment:
This should be a `require`: everything that is required for setting up the stage of the test should use `require`. When you are exercising the test subject, use `assert`.

Reply:
Thanks for the tip. Reworked.